### PR TITLE
shouldn't reference relatives that are connected in private ways

### LINKED ISSUE #346: unattached topic within topic view - unidentified bug,
unattached topic within topic view - unidentified bug, see here: http://metamaps.cc/topics/10472, http://metamaps.cc/topics/12190

### PR DESCRIPTION
**Pull request by Connoropolous**
closes #346 

@devvmh could you consult with me on this? how could we make the code more DRY? 

especially since this code is identical to the code in the Synapse policy_scope

### CODE REVIEW

**Comment by devvmh** [comment]
OK i posted a PR that... I think will fix it. You'd better review

Why is there a relatives1 and relatives2 scope anyways? Was that there all along? I'm not clear exactly on how the logic works there. Could we clean up those functions while we're mucking around in here?

Otherwise LGTM merge wheneva

**Comment by Connoropolous** [comment]
There's relatives1 and relatives2 because I didn't know how else to get all
the relatives, while also having all of them be policy_scoped. Would love
to see it happen cleaner. The code that calls this stuff that is important
is found in the show and network actions of the topic controlller

**Comment by devvmh** [comment]
Hmm so maybe it would be

```
scope :relatives, ->(topic_id = nil, user = nil) {
  synapses = Pundit.policy_scope(user, Synapse.where(node1_id: topic_id)).map(&:node1_id)
  synapses += Pundit.policy_scope(user, Synapse.where(node2_id: topic_id)).pluck(&:node2_id)
  where(id: synapses.uniq)
}
```

**Comment by Connoropolous** [comment]
Yes, I think your last comment is on point

**Comment by Connoropolous** [comment]
Except that I think the .map and .pluck have to pull the inverse node id than the one they're querying for

File: app/controllers/topics_controller.rb [hunk 1]
```diff
@@ -66,7 +64,7 @@ def relative_numbers
 
     topicsAlreadyHas = params[:network] ? params[:network].split(',').map(&:to_i) : []
 
-    @alltopics = policy_scope(Topic.relatives1(@topic.id)).to_a.concat(policy_scope(Topic.relatives2(@topic.id)).to_a).uniq
```
Comment (devvmh) [inline app/controllers/topics_controller.rb]: lines 69 and 91 used to have .uniq, but now that's inside the `relatives` scope.

**Comment by Connoropolous** [comment]
Yees :) fantastic Devin

**Pull request merged**